packages/anoncreds/src/AnonCredsApi.ts

```diff
@@ -398,13 +398,20 @@ export class AnonCredsApi {
           tailsDirectoryPath,
         })
 
-      // At this moment, tails file should be published and a valid public URL will be received
       const localTailsLocation = revocationRegistryDefinition.value.tailsLocation
 
-      const { tailsFileUrl } = await tailsFileService.uploadTailsFile(this.agentContext, {
-        revocationRegistryDefinition,
-      })
-      revocationRegistryDefinition.value.tailsLocation = tailsFileUrl
+      // A check is done whether the tails location is a URL.
+      // This state may occur when the `registerRevocationRegistryDefinition` is called
+      // once to create the TXN and once to actually submit it.
+      // When the TXN is created, it will be a file path and this code will be called.
+      // When the submission is done, it already has a URL as the location and does not
+      // need to be reuploaded
+      if (!isUri(localTailsLocation)) {
```

Comment:
I'm confused here. This `localTailsLocation` comes from the result of `this.anonCredsIssuerService.createRevocationRegistryDefinition`, which will create the object by calling anoncreds-rs `RevocationRegistryDefinition.create()` method. How is it possible for it to return an URL?

Reply:
Hmmm yeah you are correct :'). I interpreted the code a bit wrong and assumed it would just use the passed arguments, but it creates it again (which is an issue as well).

I will close this PR and open an issue for this as currently has a bug where signing the transaction by an external endorser (i.e. calling `registerRevocationRegistryDefinition` twice) does not work. The credential will contain invalid revocation data.